### PR TITLE
WPB-6258 Connection request from deleted user

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/WPB-6258

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by stefanwire — APPROVED** [review]

File: integration/test/Test/Connection.hs [hunk 1]
```diff
@@ -401,3 +402,14 @@ testFederationAllowMixedConnectWithRemote =
     connectTwoUsers alice bob
   where
     defSearchPolicy = "full_search"
+
+testPendingConnectionUserDeleted :: HasCallStack => Domain -> App ()
+testPendingConnectionUserDeleted domain = do
```
Comment (fisx) [inline integration/test/Test/Connection.hs:407]:
```suggestion
testPendingConnectionUserDeleted bobsDomain = do
```

?  I sometimes feel we're trying really hard to make identifiers mean as little as possible in /integration.  could be just me, though :)

File: services/brig/src/Brig/IO/Intra.hs [hunk 2]
```diff
@@ -288,8 +296,47 @@ notifyUserDeletionLocals ::
...
+            -- note that we switch from and to users so that the "other" user becomes the recipient of the event
+            let ucCancelled =
+                  UserConnection
+                    (qUnqualified (ucTo uc))
+                    (Qualified (ucFrom uc) d)
+                    Cancelled
+                    now
+                    (ucConvId uc)
+            let e = ConnectionUpdated ucCancelled Nothing Nothing
+            onConnectionEvent deleted conn e
```
Comment (fisx) [inline services/brig/src/Brig/IO/Intra.hs:328]: is there a reason the local and the remote part look so different?  (just curious)

File: services/brig/src/Brig/IO/Intra.hs [hunk 3]
```diff
@@ -301,23 +348,37 @@ notifyUserDeletionRemotes ::
...
-          notifyUserDeleted luidDeleted (qualifyAs uids rangedUids)
+          notifyUserDeleted luidDeleted (qualifyAs ucs ((fmap (fmap (qUnqualified . ucTo))) rangedUcs))
+          -- also sent connection cancelled events to the connections that are pending
+          let remotePendingConnections = qualifyAs ucs <$> filter ((==) Sent . ucStatus) (fromRange rangedUcs)
+          forM_ remotePendingConnections $ sendCancelledEvent luidDeleted
+
+    sendCancelledEvent :: Local UserId -> Remote UserConnection -> m ()
+    sendCancelledEvent luidDeleted ruc = do
+      runExceptT (sendConnectionAction luidDeleted Nothing (qUnqualified . ucTo <$> ruc) RemoteRescind) >>= \case
+        -- I don't think we want to abort the operation since this is running asynchronously and cannot be retried anyway
```
Comment (fisx) [inline services/brig/src/Brig/IO/Intra.hs:376]: why can't it be retried?  i'd expect this to go back to the rabbit queue and then get picked up again in a bit?

**Review by fisx — APPROVED** [review]
please ignore my comments, it's all good to go!  :)

**Review by fisx — APPROVED** [review]

**Pull request merged**